src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
+    }
+  }
```

Comment:
PullProofAI

1. The guard implementation looks solid and correctly checks the feedback secret.
2. Consider adding logging for unauthorized attempts to track potential security threats.
3. Ensure that the `Env.SECRET.FEEDBACK` is securely managed, possibly through environment variables or a secure vault.
4. It's recommended to provide more detailed error messages to differentiate between different types of errors for better debugging.
5. Overall, the guard follows the principle of least privilege by restricting access based on a specific secret.

This guard effectively ensures that only requests with a valid feedback secret are allowed to pass, enhancing the security of the system.


<!-- This is an auto-generated comment by PullProofAI -->